addons/pos_restaurant/tests/test_frontend.py

```diff
@@ -2,13 +2,15 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import odoo.tests
+from odoo import Command
 from odoo.addons.point_of_sale.tests.common_setup_methods import setup_product_combo_items
 from odoo.addons.point_of_sale.tests.common import archive_products
 from odoo.addons.point_of_sale.tests.test_frontend import TestPointOfSaleHttpCommon
+from odoo.addons.pos_online_payment.tests.online_payment_common import OnlinePaymentCommon
 
 
 @odoo.tests.tagged('post_install', '-at_install')
-class TestFrontendCommon(TestPointOfSaleHttpCommon):
+class TestFrontendCommon(TestPointOfSaleHttpCommon, OnlinePaymentCommon):
```

Comment:
The test should be made in another module. If I install `pos_restaurant` I will have `pos_online_payment` because it is automatically installed if `point_of_sale` is installed but it can happen to have `pos_restaurant` without `pos_online_payment`, in which case the test generates an error.

If you can also reproduce the error in a normal shop (you also have the floating orders) then you can put the test directly in `pos_online_payment`, otherwise it's better to put it in a module that can only be installed if both `pos_restaurant` and `pos_online_payment` are installed. I believe you can use `pos_online_payment_self_order` because it depends of `pos_online_payment` and `pos_self_order` which depends on `pos_restaurant`.